include/modem/lte_lc.h

```diff
@@ -21,282 +21,392 @@ extern "C" {
  *
  * @{
  *
- * @brief Public APIs for the LTE Link Controller.
+ * Public APIs for the LTE Link Controller.
  */
 
-/* NOTE: enum lte_lc_nw_reg_status maps directly to the registration status
- *	 as returned by the AT command "AT+CEREG?".
+/**
+ * Network registration status.
+ *
+ * @note Maps directly to the registration status as returned by the AT command `AT+CEREG?`.
  */
 enum lte_lc_nw_reg_status {
+	/** Not registered. UE is not currently searching for an operator to register to. */
 	LTE_LC_NW_REG_NOT_REGISTERED		= 0,
+
+	/** Registered, home network. */
 	LTE_LC_NW_REG_REGISTERED_HOME		= 1,
+
+	/**
+	 * Not registered, but UE is currently trying to attach or searching for an operator to
+	 * register to.
+	 */
 	LTE_LC_NW_REG_SEARCHING			= 2,
+
+	/** Registration denied. */
 	LTE_LC_NW_REG_REGISTRATION_DENIED	= 3,
+
+	/** Unknown, for example out of LTE coverage. */
 	LTE_LC_NW_REG_UNKNOWN			= 4,
+
+	/** Registered, roaming. */
 	LTE_LC_NW_REG_REGISTERED_ROAMING	= 5,
+
 	LTE_LC_NW_REG_REGISTERED_EMERGENCY	= 8,
```

Comment:
This is left without a comment on purpose. The modem never gives this status and it's not included in the AT command specification either. I'm planning to remove this in a separate PR.

Reply:
Created PR #12214 to remove `LTE_LC_NW_REG_REGISTERED_EMERGENCY`.